### ISSUE TITLE
Help please with update Synology with Docker & Unifi controller

### ISSUE THREAD

**Issue by gergelyregos** [jacobalberty/unifi-docker#534]
Hi Guys!
I need help with the latest update. So far i could backup container, and pull the latest unifi image, and then everything worked as normal. Now i'm on 6.5.55 and tried to update to 7.0.23 and when i tried to open ip:8443 i found a blank cotroller (unifi.ui.com shows controller offline when updated), like on the initial setup page.  I did not want to loose all my setup, also i have multi sites (dont want to start from scratch).
I'm a noobie on this, so i'm not familiar with command line scripts. like run it as --user. ... However i tried to run with BIND_PRIV false or true; RUNAS_UID false & true it dows not matter, i got blank container anyways. If i run the renamed old unifi controller (6.5.55), i can still see all my sites (at least sites managed by me), and all the devices. 
Thank you, if you can help me!

**Comment by VinnieNZ** [comment]
I've just stumbled across this issue as well.

It's not anything to do with Synology at all.

It appears to either be a missing symlink inside the docker container, or a change on the UniFi side.

Basically, according to the container documentation, all the config should be at /unifi.

However, when using 7.0+ versions of the controller, it looks like it is trying to find the config in /usr/lib/unifi.

I've gone into my controller container and verified that there is content at /usr/lib/unifi, but it doesn't match my content that is at /unifi.

What I've done is mapped the volume for the Unifi config folder that goes to /unifi to /usr/lib/unifi as well and now the container comes up fine with my config and settings all intact.

**Comment by jacobalberty** [comment]
> I've gone into my controller container and verified that there is content at /usr/lib/unifi, but it doesn't match my content that is at /unifi.

`/usr/lib/unifi` is the install folder it contains the binaries for unifi, the configuration files live under `/usr/lib/unifi/data. however

[/usr/lib/unifi/data](https://github.com/jacobalberty/unifi-docker/blob/6735259e1c3f4d41bb535f3cf7ca52e8d53b3eb5/Dockerfile#L15) should be a [symlink to /unifi/data](https://github.com/jacobalberty/unifi-docker/blob/1fde69d66a013d015fecb9f2c55bb670869194ca/docker-build.sh#L58)

```
sudo docker run --rm -i -t jacobalberty/unifi:v7.0.23 ls -al /usr/lib/unifi
[2022-03-16 02:20:31,254] <docker-entrypoint> Executing: ls -al /usr/lib/unifi
total 32
drwxr-xr-x 6 unifi unifi  4096 Feb 28 15:43 .
drwxr-xr-x 1 root  root   4096 Feb 28 15:43 ..
drwxr-xr-x 2 unifi unifi  4096 Feb 28 15:43 bin
lrwxrwxrwx 1 root  root     11 Feb 28 15:43 data -> /unifi/data
drwxr-xr-x 3 unifi unifi  4096 Feb 28 15:43 dl
drwxr-xr-x 3 unifi unifi 12288 Feb 28 15:43 lib
lrwxrwxrwx 1 root  root     10 Feb 28 15:43 logs -> /unifi/log
lrwxrwxrwx 1 root  root     10 Feb 28 15:43 run -> /unifi/run
drwxr-xr-x 3 unifi unifi  4096 Feb 28 15:43 webapps
```

**Comment by VinnieNZ** [comment]
I've removed the bind for /usr/lib/unifi altogether, so it's back to how it should be and re-run the container.

This is what is in the logs:
```
unifi    | rm: cannot remove '/var/run/unifi': Device or resource busy
unifi    | 2022/03/16 15:37:36 Chown 0:0 /unifi
unifi    | [2022-03-16 15:37:36,425] <docker-entrypoint> Starting unifi controller service.
unifi    | [2022-03-16 15:37:37,443] <docker-entrypoint> INFO: Changing 'unifi' UID to '2001' and GID to '0'
unifi    | WARN Unable to load properties from '/usr/lib/unifi/data/system.properties' - /usr/lib/unifi/data/system.properties (No such file or directory)
```

And this how /usr/lib/unifi looks for me (no data symlink):
```
docker exec -it unifi /bin/bash
root@11b8660b476f:/unifi# cd /usr/lib/unifi/
root@11b8660b476f:/usr/lib/unifi# ls -al
total 52
drwxr-xr-x 10 unifi root  4096 Mar 16 15:37 .
drwxr-xr-x  1 root  root  4096 Mar 16 15:37 ..
drwxr-xr-x  2 unifi root  4096 Mar 16 15:37 bin
drwxr-xr-x  4 unifi root  4096 Mar 16 15:38 data
drwxr-xr-x  3 unifi root  4096 Mar 16 15:37 dl
drwxr-xr-x  3 unifi root 12288 Mar 16 15:37 lib
drwxr-xr-x  2 unifi root  4096 Mar 16 15:37 logs
drwxr-xr-x  3 unifi root  4096 Mar 16 15:37 run
drwxr-xr-x  3 unifi root  4096 Mar 16 15:37 webapps
drwxr-xr-x  3 unifi root  4096 Mar 16 15:37 work
```

**Comment by jacobalberty** [comment]
How are you launching it, though it shouldn't be possible for that breakage to happen with docker command line

**Comment by VinnieNZ** [comment]
Docker-compose file.

Remember I'm using the :beta tag, so it's building each time I start the container.

**Comment by jacobalberty** [comment]
Ok, I think i can reproduce from that, op was specifically about the latest tag so I missed that you had beta

**Comment by VinnieNZ** [comment]
Awesome, thank you.

And I've just noted that in the compose file I posted above I've left the data binding in there - I added that back after I did the test where I found the symlink was missing (so I could get my controller working again).

**Comment by artomason** [comment]
Not sure if this is related at all, but going from 6.5.55 to 7.0.23 I'm receiving the message `Migrating database ...` when I navigate to the WebUI for the controller. The docker logs don't show any issues from what I can see.

**Comment by do3meli** [comment]
reference my recent comment in a closed bug about the same: https://github.com/jacobalberty/unifi-docker/issues/515#issuecomment-1083627119

**Comment by billypizer** [comment]
I was having a similar problem and found this site to solve the problem for me.
https://david.gardiner.net.au/2021/07/synology-unifi.html

**Comment by artomason** [comment]
Just tried 7.1.65, and I still cannot do the update without manual intervention.

mongodb.log:
`2022-05-23T20:12:11.413+0000 I CONTROL  [main] ***** SERVER RESTARTED *****
2022-05-23T20:12:11.424+0000 I CONTROL  [main] ERROR: Cannot write pid file to /usr/lib/unifi/run/mongod.pid: Permission denied
2022-05-23T20:12:15.550+0000 I CONTROL  [main] ***** SERVER RESTARTED *****
2022-05-23T20:12:15.561+0000 I CONTROL  [main] ERROR: Cannot write pid file to /usr/lib/unifi/run/mongod.pid: Permission denied
2022-05-23T20:12:19.632+0000 I CONTROL  [main] ***** SERVER RESTARTED *****
2022-05-23T20:12:19.644+0000 I CONTROL  [main] ERROR: Cannot write pid file to /usr/lib/unifi/run/mongod.pid: Permission denied
2022-05-23T20:12:23.721+0000 I CONTROL  [main] ***** SERVER RESTARTED *****
2022-05-23T20:12:23.732+0000 I CONTROL  [main] ERROR: Cannot write pid file to /usr/lib/unifi/run/mongod.pid: Permission denied
2022-05-23T20:12:27.804+0000 I CONTROL  [main] ***** SERVER RESTARTED *****
2022-05-23T20:12:27.815+0000 I CONTROL  [main] ERROR: Cannot write pid file to /usr/lib/unifi/run/mongod.pid: Permission denied
2022-05-23T20:12:31.899+0000 I CONTROL  [main] ***** SERVER RESTARTED *****
2022-05-23T20:12:31.910+0000 I CONTROL  [main] ERROR: Cannot write pid file to /usr/lib/unifi/run/mongod.pid: Permission denied
2022-05-23T20:12:35.986+0000 I CONTROL  [main] ***** SERVER RESTARTED *****
2022-05-23T20:12:35.997+0000 I CONTROL  [main] ERROR: Cannot write pid file to /usr/lib/unifi/run/mongod.pid: Permission denied
2022-05-23T20:13:50.030+0000 I CONTROL  [main] ***** SERVER RESTARTED *****`

server.log
`[2022-05-23T20:12:11,408] <db-server> INFO  db     - 2022-05-23T20:12:11.408+0000 I STORAGE  [main] Engine custom option: cache_size=256M
[2022-05-23T20:12:11,426] <db-server> INFO  db     - DbServer stopped
[2022-05-23T20:12:15,545] <db-server> INFO  db     - 2022-05-23T20:12:15.545+0000 I STORAGE  [main] Engine custom option: cache_size=256M
[2022-05-23T20:12:15,564] <db-server> INFO  db     - DbServer stopped
[2022-05-23T20:12:19,628] <db-server> INFO  db     - 2022-05-23T20:12:19.628+0000 I STORAGE  [main] Engine custom option: cache_size=256M
[2022-05-23T20:12:19,646] <db-server> INFO  db     - DbServer stopped
[2022-05-23T20:12:23,717] <db-server> INFO  db     - 2022-05-23T20:12:23.716+0000 I STORAGE  [main] Engine custom option: cache_size=256M
[2022-05-23T20:12:23,735] <db-server> INFO  db     - DbServer stopped
[2022-05-23T20:12:27,799] <db-server> INFO  db     - 2022-05-23T20:12:27.799+0000 I STORAGE  [main] Engine custom option: cache_size=256M
[2022-05-23T20:12:27,817] <db-server> INFO  db     - DbServer stopped
[2022-05-23T20:12:31,894] <db-server> INFO  db     - 2022-05-23T20:12:31.894+0000 I STORAGE  [main] Engine custom option: cache_size=256M
[2022-05-23T20:12:31,913] <db-server> INFO  db     - DbServer stopped
[2022-05-23T20:12:35,982] <db-server> INFO  db     - 2022-05-23T20:12:35.981+0000 I STORAGE  [main] Engine custom option: cache_size=256M
[2022-05-23T20:12:36,000] <db-server> INFO  db     - DbServer stopped`

If this is not the preferred place to put this, let me know and I will move it or open a new issue.

**Comment by valsily** [comment]
@artomason You should check @do3meli 's solution he mentioned in #515 . I had the same errors when trying to do the same upgrade on my Synology NAS and the solution works like a charm.

**Comment by artomason** [comment]
@valsily I saw that as well but my solution is a little different. The `/unifi` directory does not appear to get the correct permissions as mine was set to `999:999` even though I specified the `UNIFI_GID` and `UNIFI_UID` variables. I had to manually change the permissions by issuing `chown UID:GID /unifi`. In any case, I just wanted to report that this is still an issue so automation doesn't close the issue before we have a fix.
